# b) Encuentre una solución básica inicial factible para el PPL (ayuda: formule el problema de optimización de la Fase I de Simplex y luego encuentre el valor óptimo con Octave)

In [1]:
clear

In [2]:
%Vector de costos de función objetivo
c=[0 0 0 0 0 0 0 0 0 0 1 1]
%Matriz de restricciones
A=[1/3000 1/600 1/1000 1 0 0 0 0 0 0 0 0; 40 210 45 0 1 0 0 0 0 0 0 0; 
1 0 0 0 0 -1 0 0 0 0 1 0; 0 1 0 0 0 0 -1 0 0 0 0 1;
1 0 0 0 0 0 0 1 0 0 0 0; 0 1 0 0 0 0 0 0 1 0 0 0;
0 0 1 0 0 0 0 0 0 1 0 0]
%Vector con lado derecho de restricciones
b=[5;400000;5000;400;10000;1000;2000]
%Límite inferior variables
lb=[0 0 0 0 0 0 0 0 0 0 0 0]
%Límite superior variables
ub=[]
%Tipo de restricciones
ctype='SSSSSSS'
%Tipo de variables
vtype='CCCCCCCCCCCC'
%Opción para el tipo de output del optimizador
param.msglev=2;

c =

   0   0   0   0   0   0   0   0   0   0   1   1

A =

 Columns 1 through 6:

     0.00033     0.00167     0.00100     1.00000     0.00000     0.00000
    40.00000   210.00000    45.00000     0.00000     1.00000     0.00000
     1.00000     0.00000     0.00000     0.00000     0.00000    -1.00000
     0.00000     1.00000     0.00000     0.00000     0.00000     0.00000
     1.00000     0.00000     0.00000     0.00000     0.00000     0.00000
     0.00000     1.00000     0.00000     0.00000     0.00000     0.00000
     0.00000     0.00000     1.00000     0.00000     0.00000     0.00000

 Columns 7 through 12:

     0.00000     0.00000     0.00000     0.00000     0.00000     0.00000
     0.00000     0.00000     0.00000     0.00000     0.00000     0.00000
     0.00000     0.00000     0.00000     0.00000     1.00000     0.00000
    -1.00000     0.00000     0.00000     0.00000     0.00000     1.00000
     0.00000     1.00000     0.00000     0.00000     0.00000     0.00000
     0.00000    

In [3]:
#Ejecución del optimizador
glpk(c, A, b, lb, ub,ctype,vtype,s=1,param)

      0: obj =   0.000000000e+00 inf =   3.423e+03 (2)
      2: obj =   5.400000000e+03 inf =   0.000e+00 (0)
*     4: obj =   0.000000000e+00 inf =   0.000e+00 (0)
ans =

     5000.00000
      400.00000
        0.00000
        2.66667
   116000.00000
        0.00000
        0.00000
     5000.00000
      600.00000
     2000.00000
        0.00000
        0.00000



## Solución optima

* Función objetivo: 0  <br />
* Variables de decisión del problema original: $x_1 = 5000, x_2 = 400, x_3 = 0$ <br />
* Variales de holgura: $x_4 = 2.67, x_5 = 116000, x_6 = 0,x_7 = 0, x_8 = 5000, x_9 = 600, x_{10} = 2000$ <br />
* Variables artificales: $y_{1} = 0$, $y_{2} = 0 $

# c) Determine la solución óptima del PPL con Octave.

In [4]:
clear

In [5]:
%Vector de costos de función objetivo
c=[4000 6000 10000]  
%Matriz de restricciones
A=[1/3000 1/600 1/1000; 40 210 45;1 0 0;0 1 0;1 0 0;0 1 0;0 0 1] 
%Vector con lado derecho de restricciones
B=[5;400000;5000; 400; 10000;1000;2000]
%Límite inferior variables
lb=[0,0,0]  
%Límite superior variables
ub=[]
%Tipo de restricciones
ctype='UULLUUU'
%Tipo de variables
vartype='CCC'
%Opción para el tipo de output del optimizador
param.msglev=2;

c =

    4000    6000   10000

A =

     0.00033     0.00167     0.00100
    40.00000   210.00000    45.00000
     1.00000     0.00000     0.00000
     0.00000     1.00000     0.00000
     1.00000     0.00000     0.00000
     0.00000     1.00000     0.00000
     0.00000     0.00000     1.00000

B =

        5
   400000
     5000
      400
    10000
     1000
     2000

lb =

   0   0   0

ub = [](0x0)
ctype = UULLUUU
vartype = CCC


In [6]:
#Ejecución del optimizador
glpk(c=c,A=A,B=B,lb=lb,ub=ub,ctype=ctype,vartype=vartype,s=-1,param=param)

*     0: obj =   2.240000000e+07 inf =   0.000e+00 (3)
*     2: obj =   4.500000000e+07 inf =   0.000e+00 (0)
ans =

   5650.00
    400.00
   2000.00



## Solución optima

* Función objetivo: 45000000  <br />
* Variables de decisión: $x_1 = 5650, x_2 = 400, x_3 = 2000$